R/kernel.r

```diff
@@ -242,6 +243,75 @@ complete = function(request) {
         cursor_end = start_position + nchar(c.info$token)))
 },
 
+inspect = function(request) {
+    # 5.0 protocol:
+    code <- request$content$code
+    cursor_pos <- request$content$cursor_pos
+
+    # Define functions to add a section to content.
+    add_new_section_plain_text <- function (content, section_name, new_content) {
+        if (is.null(new_content)) return(content)
+        title <- paste0('# ', section_name, ':\n')
+        return(paste0(content, title, new_content, '\n', sep = '\n'))
+    }
+
+    add_new_section_html <- function (content, section_name, new_content) {
+        if (is.null(new_content)) return(content)
+        title <- paste0('<h1>', section_name, ':</h1>\n')
+        return(paste0(content, title, new_content, '\n', sep = '\n'))
+    }
+
+    new_section_adders <- list(
+        'text/plain' = add_new_section_plain_text,
+        'text/html' = add_new_section_html)
+
+    add_new_section <- function (data, section_name, new_data) {
+        for (mime in names(new_section_adders)) {
+            data[[mime]] <- new_section_adders[[mime]](
+                data[[mime]], section_name, new_data[[mime]])
+        }
+        return(data)
+    }
+
+    # Get token under the cursor_pos.
+    # There may be better ways to do that than this.
+    token <- ''
+    for (i in seq(cursor_pos, nchar(code))) {
+        token_candidate <- utils:::.guessTokenFromLine(code, i)
+        if (nchar(token_candidate) == 0) break
+        token <- token_candidate
+    }
+
+    data <- namedlist()
+    if (nchar(token) != 0) {
+        tryCatch(
+            {
+                obj <- eval(parse(text = token))
+
+                class_data <- IRdisplay::prepare_mimebundle(class(obj))$data
+                data <- add_new_section(data, 'Class attribute', class_data)
+
+                print_data <- IRdisplay::prepare_mimebundle(obj)$data
+                data <- add_new_section(data, 'Printed form', print_data)
+            },
+            error = function (e) NULL)
+        tryCatch(
+            {
+                help_obj <- eval(parse(text = paste0('?', token)))
```

Comment:
please also add a comment why you don’t use `help_obj <- help(obj)` here.

i assume it is because there might be no object and still a help topic, am i right?